### PR TITLE
Run the link verification step even previous step failed.

### PR DESCRIPTION
**Pull request by sima-zhu**
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=703442&view=logs&j=ad792ccc-30b2-59a1-5b53-546a79d88828&t=004b592e-cb31-5e96-4451-b7e7f6a38a50

### CODE REVIEW

File: eng/common/pipelines/templates/steps/verify-links.yml [hunk 1]
```diff
@@ -25,3 +25,4 @@ steps:
         -branchReplacementName ${{ parameters.BranchReplacementName }}
         -devOpsLogging: $true
         -checkLinkGuidance: ${{ parameters.CheckLinkGuidance }}
+    condition: succeededOrFailed()
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-links.yml:28]: We should make this a template option as opposed to doing for everyone. Please add another template parameter and only add this if that someone opts into it.

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/verify-links.yml:28]: Do we want to make succeeded() as default?

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/verify-links.yml]: I'd just exclude it if the parameter isn't set.

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/verify-links.yml]: This is $true/false value, how do we exclude it if not set?

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/verify-links.yml]:
```
 ${{ if eq(parameters.RunOnFailure, true) }}:
  condition: succeededOrFailed()
```

I believe something like that should work.

File: eng/common/pipelines/templates/steps/verify-links.yml [hunk 2]
```diff
@@ -8,10 +8,12 @@ parameters:
   Urls: '(Get-ChildItem -Path ./ -Recurse -Include *.md)'
   BranchReplaceRegex: "^(${env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI}.*/(?:blob|tree)/)master(/.*)$"
   BranchReplacementName: "${env:SYSTEM_PULLREQUEST_SOURCECOMMITID}"
+  RunOnFailure: $false
 
 steps:
   - task: PowerShell@2
     displayName: Link verification check
+    condition: or(and(succeededOrFailed(), '${{ parameters.RunOnFailure }}'), and(succeeded(), eq('${{ parameters.RunOnFailure }}', false)))
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-links.yml:16]: If my original suggestion didn't work I'd recommend we go back to your original change ad set the condition as the parameter itself with a default of `succeeded` and someone can pass in a custom condition if they want.

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/verify-links.yml:16]: Then people has to pass the `succeeded()` on top level instead of `$true` or `$false`.

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/verify-links.yml:16]: The if clause on `condition` only does not work as expected, it introduced format error.

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/verify-links.yml:16]: You can default it to that value and then if they want to run it on failure as well then the pass in `succeededOrFailed()` as the condition.

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/verify-links.yml:16]: Got you. Sure. Will have comments on how to set the parameter` condition`.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**